javascript/sdk/src/repl.ts

```diff
@@ -46,34 +51,110 @@ export type MessageFromServer =
       id: string
     }
 
+interface ReplOptions {
+  baseUrl?: string
+  token?: string
+}
+
+let createWebsocket = websocket
+
 export class Repl {
-  private ws: WebSocket
-  private listener = new EventTarget()
-  private nextRequestId = 0
-
-  constructor(ws: WebSocket) {
-    this.ws = ws
-    this.ws.addEventListener('message', ({ data }) => {
-      const msg = JSON.parse(data.toString())
-      this.listener.dispatchEvent(new CustomEvent('msg', { detail: msg }))
+  #baseUrl = 'wss://api.forevervm.com'
+  #token = process.env.FOREVERVM_TOKEN || ''
+  #machine: string | undefined
+
+  #ws: WebSocket | NodeWebSocket
+  #listener = new EventTarget()
+  #queued: MessageToServer | undefined
+  #nextRequestId = 0
+  #retries = 0
+
+  constructor(machine?: string, options?: ReplOptions)
+  constructor(options?: ReplOptions)
+  constructor(machine?: string | ReplOptions, options?: ReplOptions) {
```

Comment:
What if we just made `machine` another `ReplOption`?

Alternatively, we could have URL construction moved out of the repl and into the client, and just have `ReplOptions` take a websocket URL.

Reply:
i like the first option actually!